### ISSUE TITLE
Timing problem - brython vs js

### ISSUE THREAD

**Issue by kmindspark** [brython-dev/brython#1476]
I have these two pieces of code:
JS:
```
function my_proceed(duration){
            var start = Date.now()
            var currentTime = Date.now()
            var secondLast = 0
            while(currentTime - start < duration * 1000){
                secondLast = currentTime
                currentTime = Date.now()
                //counter += 1
            }
            if (Date.now() - start >= 41){
                console.log("internal time:", Date.now() - start)
                console.log("second last diff:", secondLast - start)
            }
}

for (var i = 0; i < 1000; i++){
   my_proceed(0.040)
}
```

Brython:
```
def proceed(duration, start=None):
    start = int(time.time() * 1000)
    current_time = int(time.time() * 1000)
    while int(round(current_time - start)) < duration*1000:
        second_last = current_time
        current_time = int(time.time() * 1000)

    if int(time.time() * 1000) - start >= 41:
        print("internal time: " + str((int(time.time() * 1000) - start)))
        print("second last diff: " + str(second_last- start))

def main():
    for i in range(1000):
        proceed(0.040)
```

For the brython code, an average output looks something like this:
```
internal time: 45
second last diff: 36
internal time: 41
second last diff: 39
```

Whereas JS outputs nothing, meaning there were no anomalies.

I'm wondering if you could replicate this difference and perhaps rectify the discrepancy between JS and Brython performance in loop iteration times? Most of the time the Brython function gets it exactly right, it's just that a very very small portion of the time, the loop seems to freeze for 5-10 ms leading to a late return from the function.

**Comment by PierreQuentel** [comment]
I ran these scripts in different situations, and I can't come to a clear conclusion.

In a page with only the JS version, there is generally no message, but I got at least one with a newly opened browser.
In a page with only the Brython version, there is always at least one message, sometimes 2 or 3.
In a page with both scripts, no message for JS, at least one for Brython.

I also ran the JS-only page in a browser tab, then opened another tab and browsed a few pages on the Internet; when the JS script finished, there were many messages (a dozen).

So yes, the Brython version "fails" more often than the Javascript version, and this is likely because of the overhead introduced by the JS code generated by Brython. But it's not as if Javascript gave the expected result 100% of the times and Brython "only" 99,5% of the times (which is the case when there are 5 messages out of 1000 tries). The cause of failure for Javascript is probably related to the execution model of a browser, with the management of the events queue.
